app/models/accounting/account.rb

```diff
@@ -16,7 +16,19 @@
 #
 
 class Accounting::Account < ActiveRecord::Base
+  QB_TRANSACTION_TYPE = 'Account'
   belongs_to :project
 
   has_many :transactions, inverse_of: :account, foreign_key: :accounting_account_id, dependent: :destroy
+
+  def self.find_or_create_from_qb_object(qb_object)
+    account = create_with(name: qb_object.name).find_or_create_by qb_id: qb_object.id
+    account.tap do |a|
+      a.update_attributes!(
+        name: qb_object.name,
+        qb_account_classification: qb_object.classification,
+        quickbooks_data: qb_object.as_json
+      )
```

Comment:
Kind of odd to set the name twice. Perhaps do `find_or_initialize_by` and then you won't need `create_with`?